### PR TITLE
[casr-ubsan] Add progress printing

### PR DESCRIPTION
**Pull request by hkctkuy**

### CODE REVIEW

File: casr/src/bin/casr-ubsan.rs [hunk 1]
```diff
@@ -111,6 +115,9 @@ fn extract_warnings(
         }
     }
 
+    // Update counter
+    *counter.write().unwrap() += 1;
```
Comment (SweetVishnya) [inline casr/src/bin/casr-ubsan.rs:119]: Counter is not increased for errors, so, `log_progress` may hang

File: casr/src/bin/casr-ubsan.rs [hunk 2]
```diff
@@ -384,19 +391,25 @@ fn main() -> Result<()> {
...
+    type Warning<'a> = (&'a PathBuf, Vec<(UbsanWarning, CrashLine)>);
+    let (warnings, _): (Vec<Warning>, _) = custom_pool.join(
+        || {
+            inputs
+                .par_iter()
+                .filter_map(|input| {
+                    let Ok(input_warnings) = extract_warnings(input, &argv, timeout, &counter) else {
+                        warn!("Failed to run program with input file {:?}", input);
+                        return None
+                    };
```
Comment (SweetVishnya) [inline casr/src/bin/casr-ubsan.rs:406]: We may increment counter just here. So, there is no need for additional `counter` argument for `extract_warnings`.

File: casr/src/bin/casr-ubsan.rs [hunk 3]
```diff
@@ -437,6 +450,13 @@ fn main() -> Result<()> {
         crashlines.len()
     );
 
+    // Rebuild thread pool (different number of threads)
+    let num_of_threads = jobs.min(inputs.len()).max(1) + 1;
```
Comment (SweetVishnya) [inline casr/src/bin/casr-ubsan.rs:454]: no need for `+ 1` here

Comment (SweetVishnya) [reply casr/src/bin/casr-ubsan.rs:454]: inputs.len() -> to_gen.len()

**Review by SweetVishnya — CHANGES_REQUESTED** [review]

File: casr/src/bin/casr-ubsan.rs (hunk 1 above)
Comment (hkctkuy) [reply casr/src/bin/casr-ubsan.rs]: Done

File: casr/src/bin/casr-ubsan.rs (hunk 2 above)
Comment (hkctkuy) [reply casr/src/bin/casr-ubsan.rs:401]: Done

File: casr/src/bin/casr-ubsan.rs (hunk 3 above)
Comment (hkctkuy) [reply casr/src/bin/casr-ubsan.rs]: Done

**Review by SweetVishnya — APPROVED** [review]

**Review by anfedotoff — APPROVED** [review]

**Pull request merged**